atxm/machine.py

```diff
@@ -39,13 +38,45 @@
 from .logging import log
 
 
-class _Machine:
+class _Machine(StateMachine):
     """
     Do not import this - it will not work.
     Please import the publicly exposed `AutomaticTxMachine` instead.
     Thanks! :)
     """
 
+    #
+    # State Machine:
+    #
+    #   Idle <---> Busy <---> Paused
+    #   | ^        | ^         | ^
+    #   V |        V |         V |
+    #    _          _           _
+    #
+    _BUSY = State("Busy")
+    _IDLE = State("Idle", initial=True)
+    _PAUSED = State("Paused")
+
+    # - State Transitions -
+    _transition_to_idle = _BUSY.to(_IDLE, unless="_busy")  # Busy -> Idle
+    _transition_to_paused = _BUSY.to(_PAUSED, cond="_pause")  # Busy -> Pause
+    _transition_to_busy = _IDLE.to(_BUSY, cond="_busy") | _PAUSED.to(
+        _BUSY, unless="_pause"
+    )  # Idle/Pause -> Busy
+    # self transitions i.e. remain in same state
+    _remain_busy = _BUSY.to.itself(cond="_busy", unless="_pause")
+    _remain_idle = _IDLE.to.itself(unless="_busy")
+    _remain_paused = _PAUSED.to.itself(cond="_pause")
```

Comment:
The state machine is strict with the transitions it allows to prevent logic errors. By default, if a transition is not defined, and it occurs the state machine will raise an exception. You can loosen this restriction to allow a state to transition to itself (self-transition) without declaring these, but for now, I'm making it explicit - but something we can revisit later. At the very least, I'm hoping to get rid of `_remain_idle` soon.

See `allow_event_without_transition ` - https://python-statemachine.readthedocs.io/en/latest/api.html#statemachine